src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -60,18 +60,26 @@ public ClientBuilder(BuildContext context)
             Dictionary<string, OperationMethod> operationMethods = new Dictionary<string, OperationMethod>(StringComparer.InvariantCultureIgnoreCase);
             foreach (Operation operation in operationGroup.Operations)
             {
-                RestClientMethod? method = BuildMethod(operation, clientName, clientParameters);
-                if (method != null)
+                int overloadCount = 0;
+                foreach (ServiceRequest serviceRequest in operation.Requests)
                 {
-                    operationMethods.Add(operation.Language.Default.Name, new OperationMethod(operation, method));
+                    HttpRequest? httpRequest = serviceRequest.Protocol.Http as HttpRequest;
+                    if (httpRequest == null)
+                    {
+                        // Only handles HTTP requests
+                        continue;
+                    }
+
+                    RestClientMethod method = BuildMethod(operation, clientName, clientParameters, httpRequest, serviceRequest.Parameters);
+                    operationMethods.Add($"{operation.Language.Default.Name}{overloadCount++}", new OperationMethod(operation, method));
```

Comment:
Wouldn't this break other scenarios? Now everything would have a suffix so https://github.com/Azure/autorest.csharp/pull/516/files#diff-657045284e38bc0dd54ff5e452fe98eaR92 wouldn't work.

Reply:
Yep, 

```
FATAL: System.Exception: The x-ms-pageable operationName "Paging_NextFragment" for operation Paging_GetMultiplePagesFragmentNextLink was not found.
```

Reply:
Fixed.